src/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -147,6 +147,12 @@ function discoverController($scope, config, courier, $route, $window, Notifier,
   .highlightAll(true)
   .version(true);
 
+  // we used to by default always add in the timeFieldName implicitly, so we're
+  // automatically adding this in for the user for legacy savedSearches
+  if (!savedSearch.explicitTimeColumn && $scope.indexPattern.timeFieldName) {
```

Comment:
`!savedSearch.explicitTimeColumn` is true for new saved searches as well, which means they never get the default columns further down: `savedSearch.columns.length > 0 ? savedSearch.columns : defaultColumns`. That is probably not intended, is it?